packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/events/KModuleEventEmitterWrapper.kt

```diff
@@ -23,22 +25,27 @@ class KModuleEventEmitterWrapper(
 ) : KEventEmitterWrapper(legacyEventEmitter, reactContextHolder) {
   override fun emit(eventName: String, eventBody: Bundle?) {
     checkIfEventWasExported(eventName)
-    super.emit(eventName, eventBody)
+    emitNative(eventName, eventBody?.toJSValue(JSTypeConverter.DefaultContainerProvider) as? ReadableNativeMap)
   }
 
   override fun emit(eventName: String, eventBody: WritableMap?) {
     checkIfEventWasExported(eventName)
-    super.emit(eventName, eventBody)
+    emitNative(eventName, eventBody as? ReadableNativeMap)
   }
 
   override fun emit(eventName: String, eventBody: Record?) {
     checkIfEventWasExported(eventName)
-    super.emit(eventName, eventBody)
+    emitNative(eventName, eventBody?.toJSValue(JSTypeConverter.DefaultContainerProvider) as? ReadableNativeMap)
   }
 
   override fun emit(eventName: String, eventBody: Map<*, *>?) {
     checkIfEventWasExported(eventName)
-    super.emit(eventName, eventBody)
+    emitNative(eventName, eventBody?.toJSValue(JSTypeConverter.DefaultContainerProvider) as? ReadableNativeMap)
+  }
+
+  private fun emitNative(eventName: String, eventBody: ReadableNativeMap?) {
```

Comment:
Events can now take more than one argument (and of any type)

Reply:
Currently, the modules API has a limit on the number of items that can be sent. However, I plan to extend it soon.